### PR TITLE
surface: linux 6.6.13 -> 6.7.6

### PR DESCRIPTION
**Pull request by gitwah**
###### Things done

- [ ] Tested the changes in your own NixOS Configuration
- [ ] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by onny** [comment]
Fixes https://github.com/NixOS/nixos-hardware/issues/877

**Comment by mexisme** [comment]
> Is it possible to build the kernel locally from inside the nixos-hardware repository? @stigtsp
> 
> this didn't work
> 
> ```
> nix-build -E "(import <nixpkgs> {}).callPackage ./microsoft/surface/common/kernel/linux-pac
> kage.nix {}"
> ```

Because `nixos-hardware` tends to be trying to provide config at the machine-level, I've found it tricky to avoid running a build+install on real hardware.

Running the [test suite](https://github.com/NixOS/nixos-hardware/blob/master/CONTRIBUTING.md#3-testing) def. catches things like syntax errors, but I personally also try to use `nixos-rebuild build-vm` on a faster machine, against a copy of the MS Surface's Nixos config, to properly test it.

If it builds at that stage, it's (so far) always been good enough to try installing on the Surface itself — which you'll ultimately _have_ to do to check the kernel changes are working, of course!

File: microsoft/surface/common/kernel/default.nix [hunk 1]
```diff
@@ -5,7 +5,7 @@ let
 
 in {
   imports = [
-    ./linux-6.6.x
+    ./linux-6.7.x
```
Comment (mexisme) [inline microsoft/surface/common/kernel/default.nix:8]: If you're not going to expose the 6.6.x `.nix` files, you may as well delete them, yeah?

In the past, @Mic92 has asked people to try to minimise the number of different kernel versions that have to be maintained for any particular hardware.

Comment (Mic92) [reply microsoft/surface/common/kernel/default.nix:8]: Yeah, sound like a good idea.

**Comment by mexisme** [comment]
I'll try to build + test this on my Surface Go tonight (GMT+12)

File: microsoft/surface/common/kernel/linux-6.7.x/default.nix [hunk 2]
```diff
@@ -7,14 +7,14 @@ let
...
-  version = "6.6.13";
+  version = "6.7.6";
   kernelPatches = surfacePatches {
     inherit version;
     patchFn = ./patches.nix;
   };
   kernelPackages = linuxPackage {
     inherit version kernelPatches;
-    sha256 = "sha256-iLiefdQerU46seQRyLuNWSV1rPgVzx3zwNxX4uiCwLw=";
+    sha256 = "";
```
Comment (tracteurblinde) [inline microsoft/surface/common/kernel/linux-6.7.x/default.nix:17]: Empty hash should be `sha256-5InsDhNw0Im0RtVlre16aYCT0rfEEioY8h7bbvk9N9M=`

**Comment by iwanders** [comment]
> Because nixos-hardware tends to be trying to provide config at the machine-level, I've found it tricky to avoid running a build+install on real hardware.

@mexisme , @onny  Just though I'd share my workflow; what I do is from my [flake](https://github.com/iwanders/nixos-surface);
```
nix build .#nixosConfigurations.<hostname>.config.system.build.toplevel -L
```
Then I push that result to my surface with `nix copy --to "ssh://<username>@<hostname>" ./result`. That allows me to build the kernel and other packages on my desktop at least, verifying it runs on the device is still necessary.

File: microsoft/surface/common/kernel/linux-6.7.x/default.nix (hunk 2 above)
Comment (iwanders) [reply microsoft/surface/common/kernel/linux-6.7.x/default.nix:17]: Any chance we can bump this to `arch-6.7.9-1`? That pulls in https://github.com/linux-surface/linux-surface/commit/852fcb1c2f7be2ac8e115984432761e0edaa7801 which has improvements to the fan profiles on the SP9.

To cherry-pick (on top of @tracteurblinde 's commit); cf462d6d1bb8d3c023ccabdb56e5c0af1a1802ed

Comment (Yusuf-Duran) [reply microsoft/surface/common/kernel/linux-6.7.x/default.nix:17]: > Any chance we can bump this to `arch-6.7.9-1`? That pulls in [linux-surface/linux-surface@852fcb1](https://github.com/linux-surface/linux-surface/commit/852fcb1c2f7be2ac8e115984432761e0edaa7801) which has improvements to the fan profiles on the SP9.
> 
> To cherry-pick (on top of @tracteurblinde 's commit); [cf462d6](https://github.com/NixOS/nixos-hardware/commit/cf462d6d1bb8d3c023ccabdb56e5c0af1a1802ed)

not only that but also finally has working cameras again which would be nice to have

**Comment by stigtsp** [comment]
Since 6.7 is EOL, can this PR be updated to 6.8? I'm happy to test on my Surfaces :)

**Comment by Mic92** [comment]
I think this was superseeded by https://github.com/NixOS/nixos-hardware/pull/912/files

**Pull request closed without merging**